src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs

```diff
@@ -3475,10 +3481,10 @@ public async Task UploadArchiveToGithubStorage_Should_Upload_Stream_Content()
         var expectedUri = "gei://archive/123456";
         var jsonResponse = $"{{ \"uri\": \"{expectedUri}\" }}";
 
-        _githubClientMock
-            .Setup(m => m.PostAsync(It.IsAny<string>(), It.IsAny<StreamContent>(), null))
-            .ReturnsAsync(jsonResponse);
-
+        // Mocking the Upload method on _archiveUploader to return the expected URI
+        _archiveUploader
+            .Setup(m => m.Upload(It.IsAny<Stream>(), archiveName, orgDatabaseId))
```

Comment:
Let's be explicit about what we are expecting to receive:

```suggestion
            .Setup(m => m.Upload(archiveContent, archiveName, orgDatabaseId))
```